x-pack/plugins/security_solution/public/common/components/top_n/top_n.tsx

```diff
@@ -101,13 +98,12 @@ const TopNComponent: React.FC<Props> = ({
     () => (
       <ViewSelect
         data-test-subj="view-select"
-        disabled={!isDetectionsAlertsTable(scopeId)}
```

Comment:
When this select only has one item (i.e. in Timeline), nothing happens if the user reselects it, per the gif below:

![select_with_one_item](https://user-images.githubusercontent.com/4459398/198360080-da76fa20-4b37-47f6-bbf3-2713c7b27ed9.gif)

_Above: Reselecting the (only) option has no effect_

Consider something like

```ts
        disabled={options.length === 1}
```

to disable the select when it has only one option.

Reply:
Disabling drop down on timeline (even though it only has 1 drop down) is flagged on  https://github.com/elastic/kibana/issues/126040
In such case, what is the expected behavior?
![image](https://user-images.githubusercontent.com/18648970/198383197-d5c8dfe2-6042-456d-b386-a455296b9b5e.png)

Reply:
> Disabling drop down on timeline (even though it only has 1 drop down) is flagged on https://github.com/elastic/kibana/issues/126040 In such case, what is the expected behavior?

[Issue 126040](https://github.com/elastic/kibana/issues/126040) does note that the dropdown is disabled in Timeline, however that is the expected behavior. Here's the background:

Timeline is the only data table in the Security Solution that mixes alerts and events. An investigation may, for example, start with an alert, and then quickly widen in scope to include events.

When a table displays only alerts, like the Alerts page, the Top N dropdown defaults to alerts, because we want to (by default) query the same dataset that's being visualized in the table. When a table displays only events, (e.g. Hosts > Events), the dropdown defaults to events for the same reason.

For Timeline, neither of the two defaults for the Top N dropdown (’just alerts” or “just events”) will work in every scenario, because at any given moment, Timeline may be displaying just alerts, just events, or both. The mix of alerts and events may change implicitly when a query is (re)executed, or explicitly via the `Show only detection alerts` toggle in the Data view selector.

Given that neither the "just alerts" nor "just events" defaults work for Timeline, and the mix of data may change at any moment, the Top N dropdown in Timeline has always:
- been disabled
- visualized both alerts and events, though the description was confusing

Before the improvement in this PR, the value shown in Timeline's disabled Top N dropdown, `All events` was confusing, because the results included both alerts and events, per the following screenshot of an inspected Timeline Top N query:

![inspect_timeline_topn_query](https://user-images.githubusercontent.com/4459398/198679371-18b59530-3be6-47a5-9d69-77c84f2d63a9.png)

_Above: Inspecting a Timeline Top N query in `main`_

Timeline is the only context in the Security Solution where the Top N dropdown is expected:
- to be disabled
- to show both alerts and events

For completeness, it's worth mentioning that the other context where [Issue 126040](https://github.com/elastic/kibana/issues/126040) mentions a disabled dropdown, an alert attached to a case, was indeed identifying a bug:

❌ The dropdown should NOT have been disabled for the alert attached to the case
❌ The default for that dropdown, events, was also incorrect, because alert data was being visualized

While desk testing, I confirmed this PR fixes the above:

✅  The dropdown is enabled for the alert attached to the case
✅ The default for that dropdown is alerts